system-x/services/flink/src/main/java/software/tnb/flink/resource/openshift/OpenshiftFlinkImage.java

```diff
@@ -0,0 +1,310 @@
+        WaitUtils.waitFor(() -> HTTPUtils.getInstance().get(externalHostname(), false).isSuccessful()
+            , 5, 5000L
+            , "Waiting until the Flink API route is ready");
+    }
+
+    @Override
+    public void closeResources() {
+        if (apiRoute != null) {
+            OpenshiftClient.get().routes().resource(apiRoute).delete();
+        }
+    }
+
+    @Override
+    public void create() {
+        sccName = "tnb-flink-" + OpenshiftClient.get().getNamespace();
+
+        serviceAccountName = name() + "-sa";
+
+        OpenshiftClient.get().serviceAccounts()
+            .createOrReplace(new ServiceAccountBuilder()
+                .withNewMetadata()
+                  .withName(serviceAccountName)
+                .endMetadata()
+            .build()
+        );
+
+        OpenshiftClient.get().addUsersToSecurityContext(
+            OpenshiftClient.get().createSecurityContext(sccName, "anyuid", "SYS_CHROOT"),
+            OpenshiftClient.get().getServiceAccountRef(serviceAccountName));
+
+        LOG.info("Deploying Flink server");
+
+        createDeployments(nameJobManager(), "jobmanager");
+
+        createJobManagerService();
+
+        createRestService();
+
+        WaitUtils.waitFor(() -> {
+            return filterPods(nameJobManager()) != null && this.isReady(nameJobManager());
+        }, "Waiting for pod ready");
+
+        createDeployments(nameTaskManager(), "taskmanager");
+
+        createTMQueryStateService();
+
+        WaitUtils.waitFor(() -> {
+            return filterPods(nameTaskManager()) != null && this.isDeployed();
+        }, "Waiting for pod ready");
+
+    }
+
+    private void createDeployments(String containerName, String command) {
+        OpenshiftClient.get().apps().deployments().createOrReplace(
+            new DeploymentBuilder()
+                .withNewMetadata()
+                    .withName(containerName)
+                    .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                    .addToLabels("component", containerName)
+                    .addToAnnotations("openshift.io/scc", sccName)
+                .endMetadata()
+                .editOrNewSpec()
+                .editOrNewSelector()
+                    .addToMatchLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                    .addToMatchLabels("component", containerName)
+                .endSelector()
+                .withReplicas(1)
+                .editOrNewTemplate()
+                    .editOrNewMetadata()
+                        .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                        .addToLabels("component", containerName)
+                    .endMetadata()
+                    .editOrNewSpec()
+                        .withServiceAccount(serviceAccountName)
+                        .addNewContainer()
+                            .withName(containerName)
+                            .withImage(defaultImage())
```

Comment:
we should use `image()` instead of `defaultImage()` to allow the image to be configurable

Reply:
Even if already implemented in the previous version, this is updated with the correct method call now